hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/stats/SidecarInstrumentation.java

```diff
@@ -0,0 +1,174 @@
+    void addSample(@NonNull final SidecarType type, long size, @NonNull final Duration duration);
+
+    /**
+     * Add the entire accumulated values of a SidecarInstrumentation to this one (which is a summary
+     * of multiple records
+     */
+    void addSamples(@NonNull final SidecarInstrumentation samples);
+
+    /**
+     * Start timing a split. Use the returned record in a try-with-resources that encloses the
+     * computation to be timed.
+     */
+    Closeable startTimer(@NonNull final SidecarType type);
+
+    /**
+     * Measure a duration split for a computation given by a `Runnable`, add that duration split as
+     * a sample.
+     */
+    void captureDurationSplit(@NonNull final SidecarType type, @NonNull final Runnable fn);
+
+    /** Just like a callable except _doesn't_ declare it throws a checked exception */
+    interface ExceptionFreeCallable<V> {
+        V call();
+    }
+
+    /**
+     * Measure a duration split for a computation given by a `Runnable`, add that duration split as
+     * a sample.
+     */
+    <V> V captureDurationSplit(
+            @NonNull final SidecarType type, @NonNull final ExceptionFreeCallable<V> fn);
+
+    /**
+     * Finalize a duration measurement by adding all the splits, for one sidecar type, together as
+     * one sample to the compute duration.
+     */
+    void addDurationSplitsAsDurationSample(@NonNull final SidecarType type);
+
+    /* Reset all accumulators (all sidecar types) to zero samples */
+    void reset();
+
+    // Copy management: the same measurements are used for multiple sidecar transactions.  And
+    // they need to be passed across thread boundaries via a queue.  There's no way to know which
+    // is the last one for a given sidecar record, and no way to signal that across the queue.
+    // So a counter is incremented every time this measurement set is passed through the queue,
+    // and decremented when it is removed.  When the counter reaches zero it's time to log and
+    // report
+    // metrics.
+
+    /** Increment the copy counter when passing this across the record stream queue. */
+    void addCopy();
+
+    /**
+     * Decrement the copy counter when processing this after fetching it from the records stream
+     * queue. Returns `true` when this is the _last_ copy coming across the queue (it's time to log
+     * and issue metrics).
+     */
+    boolean removeCopy();
+
+    String toString(@NonNull final SidecarType type);
+
+    /**
+     * Create a _cheap_ (no heap!) do-nothing implementation for: 1. some uses where a
+     * instrumentation is required by a function signature but we don't need the metrics 2. some
+     * uses where you need to create a placeholder before a "real" one is created 3. all the tests
+     * where we aren't tracking metrics at all
+     */
+    @SuppressWarnings(
+            "java:S1186") // suppress warning about lots of empty method bodies: it's intentional
+    static SidecarInstrumentation createNoop() {
```

Comment:
Are you referring to the suppress or to the `createNoop` method itself?  (And either way, is there some record of those discussions I could read?)

Reply:
The Noop implementation.

Reply:
Using a noop-implementation in such cases has advantages and disadvantages. It is probably a matter of personal preference, if you want to use them. As I once wasted countless hours because of a noop-implementation, I tend to focus on the downsides. 😄

Reply:
All but one (I think, I'll check) uses are in tests - so except for that one I could move this over to the tests, and contain it that way.  For that one, which is only for the migration scenario where it is not needed (since we don't have to measure anything for the one-time migration scenario) I'll look again at what I can do.

Reply:
I changed my mind about moving the noop impl over to the tests: that's too far away.  Putting it as a peer.  (And the factory method `createNoop()` in the interface works! I'm astonished to find out this kind of circular-importing _works_! Java compiler/classloader FTW!)

I do hear what you're saying about the potential downside.  I just spent some time debugging where the problem was a _mock_ that failed to implement something new.